### PR TITLE
fixes #2601: AddDefaultODataServices should pass OData version to low layer

### LINKED ISSUE #2601: AddDefaultODataServices should pass OData version to lower layer
*Short summary (3-5 sentences) describing the issue.*

### Assemblies affected

*Which assemblies and versions are known to be affected e.g. OData .Net lib 7.x*

### Reproduce steps

```C#
public static IContainerBuilder AddDefaultODataServices(this IContainerBuilder builder, ODataVersion odataVersion)
```

has the ODataVersion argument.

this ODataVersion is passed to ODataSimplifiedOptions, but it should also be passed to others.

for example: ODataMessageReaderSettings

### Expected result

*What would happen if there wasn't a bug.*

### Actual result

*What is actually happening.*

### Additional detail

*Optional, details of the root cause if known. Delete this section if you have no additional details to add.*

### PR DESCRIPTION
**Pull request by xuzhg**
…er layer

### Issues

*This pull request fixes #2601 .*

### Description

* Pass OData version to lower layer in AddDefaultODataServices

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Review by gathogojr — DISMISSED** [review]

File: src/Microsoft.OData.Core/ContainerBuilderExtensions.cs [hunk 1]
```diff
@@ -136,7 +136,7 @@ public static IContainerBuilder AddDefaultODataServices(this IContainerBuilder b
             builder.AddService<IJsonWriterFactoryAsync>(ServiceLifetime.Singleton, sp => new DefaultJsonWriterFactory());
             builder.AddService(ServiceLifetime.Singleton, sp => ODataMediaTypeResolver.GetMediaTypeResolver(null));
             builder.AddService<ODataMessageInfo>(ServiceLifetime.Scoped);
-            builder.AddServicePrototype(new ODataMessageReaderSettings());
+            builder.AddServicePrototype(new ODataMessageReaderSettings(odataVersion));
             builder.AddService(ServiceLifetime.Scoped, sp => sp.GetServicePrototype<ODataMessageReaderSettings>().Clone());
             builder.AddServicePrototype(new ODataMessageWriterSettings());
```
Comment (ElizabethOkerio) [inline src/Microsoft.OData.Core/ContainerBuilderExtensions.cs:141]: should it be passed to the writer settings too?

Comment (xuzhg) [reply src/Microsoft.OData.Core/ContainerBuilderExtensions.cs:141]: ODataMessageWriterSettings doesn't have a constructor to accept ODataVersion.

Comment (ElizabethOkerio) [reply src/Microsoft.OData.Core/ContainerBuilderExtensions.cs:141]: How will the version be passed to the writer settings?

**Review by ElizabethOkerio — DISMISSED** [review]

File: src/Microsoft.OData.Core/ContainerBuilderExtensions.cs (hunk 1 above)
Comment (ElizabethOkerio) [reply src/Microsoft.OData.Core/ContainerBuilderExtensions.cs:141]: or why does the reader settings have a constructor that accepts an ODataVersion but the writer settings doesn't.

**Review by KenitoInc — DISMISSED** [review]

File: src/Microsoft.OData.Core/ContainerBuilderExtensions.cs (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.OData.Core/ContainerBuilderExtensions.cs:141]: I have the same concern. Maybe @mikepizzo  has some input.

Comment (gathogojr) [inline src/Microsoft.OData.Core/ContainerBuilderExtensions.cs:141]: @xuzhg Can we do it this way for `ODataMessageWriterSettings`?
```suggestion
            builder.AddServicePrototype(new ODataMessageWriterSettings { Version = odataVersion });
```

**Pull request merged**